backend/services/gemini_service.py

```
@@ -222,3 +222,81 @@ def analyze_bug_report(self, bug_title: str, description: str, steps: str, error
         except Exception as e:
             print(f"[GeminiService] Bug Analysis Error: {e}")
             return f"Diagnostic analysis failed: {str(e)}"
+
+    def ask_ai(self, prompt: str, ticket_context: dict, history: list, image_base64: str = None) -> str:
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)